src/Models/RaspberryPiDevice.ts

```diff
@@ -74,118 +77,164 @@ export class RaspberryPiDevice implements Device {
 
-    if (sketchFileName === undefined) {
-      return false;
-    } else if (!sketchFileName) {
-      sketchFileName = constants.defaultSketchFileName;
-    } else {
-      sketchFileName = sketchFileName.trim();
-      if (!/\.js$/i.test(sketchFileName)) {
-        sketchFileName += '.js';
-      }
+    if (fs.existsSync(cppPropertiesFilePath)) {
+      return true;
     }
 
-    const sketchTemplateFilePath =
-        this.extensionContext.asAbsolutePath(path.join(
-            FileNames.resourcesFolderName, RaspberryPiDevice.boardId,
-            this.sketchName));
-    const newSketchFilePath = path.join(deviceFolderPath, sketchFileName);
-
     try {
-      const content = fs.readFileSync(sketchTemplateFilePath).toString();
-      fs.writeFileSync(newSketchFilePath, content);
+      const propertiesSourceFile = path.join(
+        this.boardFolderPath, RaspberryPiDevice.boardId, FileNames.cppPropertiesFileName);
+      const propertiesContent =
+          fs.readFileSync(propertiesSourceFile).toString();
+      fs.writeFileSync(cppPropertiesFilePath, propertiesContent);
     } catch (error) {
-      throw new Error(`Create ${sketchFileName} failed: ${error.message}`);
+      throw new Error(`Create cpp properties file failed: ${error.message}`);
     }
 
-    const packageTemplateFilePath =
-        this.extensionContext.asAbsolutePath(path.join(
-            FileNames.resourcesFolderName, RaspberryPiDevice.boardId,
-            'package.json'));
-    const newPackageFilePath = path.join(deviceFolderPath, 'package.json');
+    return true;
+  }
 
-    try {
-      const packageObj = require(packageTemplateFilePath);
-      packageObj.main = sketchFileName;
-      fs.writeFileSync(newPackageFilePath, JSON.stringify(packageObj, null, 2));
-    } catch (error) {
-      throw new Error(`Create package.json failed: ${error.message}`);
-    }
+  async generateDockerRelatedFiles(): Promise<boolean> {
+        // Dockerfile       
+        const dockerfileTargetPath = path.join(
+          this.devcontainerFolderPath, FileNames.dockerfileName);
 
-    const settingsJSONFilePath =
-        path.join(vscodeFolderPath, FileNames.settingsJsonFileName);
-    const settingsJSONObj = {'files.exclude': {'.iotworkbenchproject': true}};
+        if (fs.existsSync(dockerfileTargetPath)) {
+          return true;
+        }
 
-    try {
-      fs.writeFileSync(
-          settingsJSONFilePath, JSON.stringify(settingsJSONObj, null, 4));
-    } catch (error) {
-      throw new Error(`Device: create config file failed: ${error.message}`);
-    }
+        try {
+          const dockerfileSourcePath = path.join(
+            this.boardFolderPath, RaspberryPiDevice.boardId, FileNames.dockerfileName);
+          const dockerfileContent = fs.readFileSync(dockerfileSourcePath, 'utf8');
+          fs.writeFileSync(dockerfileTargetPath, dockerfileContent);
+        } catch (error) {
+          throw new Error(`Create Dockerfile failed: ${error.message}`);
+        }

```

Comment:
Similar pieces in AZ3166Device.ts and RaspberryPiDevice.ts
Consider moving this to a common function, e.g.:

async generateFile(board: Board, targetPath: string, sourceFaileName: string): Promise<boolean>